### ISSUE TITLE
Applicable driver not found; attempting to install with Selenium Manager (Beta)

### ISSUE THREAD

**Issue by Habeeb556** [apache/superset#29734]
I got this INFO in celery log:

`[WARNING/ForkPoolWorker-3] Applicable driver not found; attempting to install with Selenium Manager (Beta)`

_Originally posted by @Habeeb556 in https://github.com/apache/superset/issues/29401#issuecomment-2228313179_

**Comment by Habeeb556** [comment]
@dosu I am not using docker.

**Comment by Habeeb556** [comment]
@dosu I am not using a docker

**Comment by Habeeb556** [comment]
@dosu find the below now:

```
[WARNING/ForkPoolWorker-1] Failed at generating thumbnail Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line
Traceback (most recent call last):
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/superset/utils/screenshots.py", line 170, in compute_and_cache
   payload = self.get_screenshot(user=user, window_size=window_size)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/superset/utils/screenshots.py", line 93, in get_screenshot
   self.screenshot = driver.get_screenshot(self.url, self.element, user)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/superset/utils/webdriver.py", line 354, in get_screenshot
   driver = self.auth(user)
            ^^^^^^^^^^^^^^^
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/superset/utils/webdriver.py", line 276, in auth
   driver = self.create()
            ^^^^^^^^^^^^^
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/superset/utils/webdriver.py", line 273, in create
   return driver_class(**kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/selenium/webdriver/firefox/webdriver.py", line 201, in __init__
   super().__init__(command_executor=executor, options=options, keep_alive=True)
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 286, in __init__
   self.start_session(capabilities, browser_profile)
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 378, in start_session
   response = self.execute(Command.NEW_SESSION, parameters)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
   self.error_handler.check_response(response)
 File "/swloc/.virtualenvs/supersetvenv4/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
   raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line

```

**Comment by Habeeb556** [comment]
To conclude this thread error with Selenium:

All I need to do is [install geckodriver manually](https://github.com/apache/superset/issues/29401#issuecomment-2228362893) since the $PATH automatically finds `/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin`.

Then, install Firefox using:

```bash
sudo yum install firefox
```

This should resolve the following issue:

```plaintext
selenium.common.exceptions.SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line
```

as suggested [here](https://github.com/mozilla/geckodriver/issues/1344).

However, I've encountered another issue now:

```
selenium.common.exceptions.WebDriverException: Message: Reached error page: about:neterror?e=connectionFailure&u=http%3A//0.0.0.0%3A8080/login/&c=UTF-8&d=Firefox%20can%E2%80%99t%20establish%20a%20connection%20to%20the%20server%20at%200.0.0.0%3A8080.
Stacktrace:
RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:187:5
UnknownError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:736:5
checkReadyState@chrome://remote/content/marionette/navigate.sys.mjs:61:24
onNavigation@chrome://remote/content/marionette/navigate.sys.mjs:329:39
emit@resource://gre/modules/EventEmitter.sys.mjs:154:20
receiveMessage@chrome://remote/content/marionette/actors/MarionetteEventsParent.sys.mjs:35:25
```

@dosu

**Comment by Habeeb556** [comment]
Resolved with #9135

change `WEBDRIVER_BASEURL = "http://0.0.0.0:8080/"` to `WEBDRIVER_BASEURL = "http://0.0.0.0:8088/"` and add this to `superset_config.py`

Then, faced another error:

`[2024-07-30 16:27:50,021: ERROR/ForkPoolWorker-4] Selenium timed out waiting for chart containers to draw at url http://0.0.0.0:8088/superset/dashboard/24/?standalone=3`

And I think it is related to #29730 and resolved with this https://github.com/apache/superset/issues/29401#issuecomment-2228250274

```
SCREENSHOT_LOCATE_WAIT = 200  # Increase the wait time for locating elements
SCREENSHOT_LOAD_WAIT = 1200   # Increase the wait time for loading the page
```
